node/init.go

```diff
@@ -7,23 +7,30 @@ import (
 	"github.com/lightstreams-network/lightchain/consensus"
 	"github.com/lightstreams-network/lightchain/database"
 	"github.com/lightstreams-network/lightchain/setup"
+	
+	"github.com/lightstreams-network/lightchain/tracer"
 )
 
 func Init(cfg Config, ntw setup.Network) error {
 	logger := log.NewLogger().With("engine", "node")
 	logger.Info("Initializing lightchain node data dir...", "dir", cfg.DataDir)
+	
+	trc, err := tracer.NewTracer(cfg.tracerCfg)
+	if err != nil {
+		return err
+	}
 
 	if err := os.MkdirAll(cfg.DataDir, os.ModePerm); err != nil {
 		return err
 	}
 	
-	if err := consensus.Init(cfg.consensusCfg, ntw); err != nil {
+	if err := consensus.Init(cfg.consensusCfg, ntw, &trc); err != nil {
```

Comment:
why pointer?

Reply:
I left that behind, I tried several approaches and one of them was modifying the tracer